pkgs/development/python-modules/llama-index/default.nix

```diff
@@ -0,0 +1,65 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pytestCheckHook
+, python
+, pydantic
+, dataclasses-json
+, httpx
+, openai
+, deprecated
+, tenacity
+, numpy
+, beautifulsoup4
+, pandas
+, fsspec
+, nest-asyncio
+, sqlalchemy
+, transformers
+, torch
+, nltk
+, aiostream
+}:
+
+buildPythonPackage rec {
+  pname = "llama-index";
+  version = "0.9.13";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "run-llama";
+    repo = "llama_index";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-sXG9yhls3zPHGYCSCJMDp4T02O4kCDp7pr40aWEplt8=";
+  };
+
+  nativeBuildInputs = [ python.pkgs.poetry-core ];
+
+  propagatedBuildInputs = [
```

Comment:
Could you please add the exact dependencies?
https://github.com/run-llama/llama_index/blob/b4e7537eacceffa36c0931a15df3fad311997a3a/pyproject.toml#L44

Also, please consider adding optional-dependencies.